### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/dnannsum`

### PR DESCRIPTION
**Pull request by headlessNode**
## Description

> What is the purpose of this pull request?

This pull request:

-   add C `ndarray` API and refactor/cleanup `blas/ext/base/dnannsum`

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by headlessNode** [comment]
@kgryte not ready for review but can I get insight into why `run_test_coverage` CI is failing when `tests` are fine?

**Comment by kgryte** [comment]
@headlessNode We are not sure why. @Planeshifter is currently working on a fix, as we have been seeing this across multiple PRs.

File: lib/node_modules/@stdlib/blas/ext/base/dnannsum/examples/c/example.c [hunk 1]
```diff
@@ -17,27 +17,26 @@
...
 	// Create a strided array:
 	const double x[] = { 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 0.0/0.0, 0.0/0.0 };
 
 	// Specify the number of elements:
-	const int64_t N = 5;
+	const int N = 5;
 
 	// Specify the stride length:
-	const int64_t stride = 2;
+	const int stride = 2;
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsum/examples/c/example.c]:
```suggestion
	const int strideX = 2;
```

File: lib/node_modules/@stdlib/blas/ext/base/dnannsum/examples/c/example.c [hunk 2]
```diff
@@ -17,27 +17,26 @@
...
 
 	// Initialize a variable for storing the number of non-NaN elements:
-	int64_t n = 0;
+	CBLAS_INT n = 0;
 
 	// Compute the sum:
 	double v = stdlib_strided_dnannsum( N, x, stride, &n );
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dnannsum/examples/c/example.c]:
```suggestion
	double v = stdlib_strided_dnannsum( N, x, strideX, &n );
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @headlessNode!

**Pull request merged**